extensions/panache/hibernate-reactive-panache-common/deployment/src/main/java/io/quarkus/hibernate/reactive/panache/common/deployment/PanacheJpaCommonResourceProcessor.java

```diff
@@ -132,27 +111,46 @@ void transformResourceMethods(CombinedIndexBuildItem index, Capabilities capabil
                     DotName.createSimple("jakarta.ws.rs.PATCH"), DotName.createSimple("jakarta.ws.rs.POST"),
                     DotName.createSimple("jakarta.ws.rs.PUT"));
-            List<DotName> bindings = List.of(REACTIVE_TRANSACTIONAL, WITH_SESSION, WITH_SESSION_ON_DEMAND, WITH_TRANSACTION);
+            List<DotName> bindings = List.of(DotNames.REACTIVE_TRANSACTIONAL, DotNames.WITH_SESSION,
+                    DotNames.WITH_SESSION_ON_DEMAND, DotNames.WITH_TRANSACTION);
 
-            // Collect all panache entities
+            // Collect all Panache entities and repositories
             Set<DotName> entities = new HashSet<>();
-            for (ClassInfo subclass : index.getIndex().getAllKnownSubclasses(PANACHE_ENTITY_BASE)) {
-                if (!subclass.name().equals(PANACHE_ENTITY)) {
+            for (ClassInfo subclass : index.getIndex().getAllKnownSubclasses(DotNames.PANACHE_ENTITY_BASE)) {
+                if (!subclass.name().equals(DotNames.PANACHE_ENTITY)) {
                     entities.add(subclass.name());
                 }
             }
-            for (ClassInfo subclass : index.getIndex().getAllKnownImplementors(PANACHE_KOTLIN_ENTITY_BASE)) {
-                if (!subclass.name().equals(PANACHE_KOTLIN_ENTITY)) {
-                    entities.add(subclass.name());
+            for (ClassInfo implementor : index.getIndex().getAllKnownImplementors(DotNames.PANACHE_KOTLIN_ENTITY_BASE)) {
+                if (!implementor.name().equals(DotNames.PANACHE_KOTLIN_ENTITY)) {
+                    entities.add(implementor.name());
                 }
             }
-            Set<DotName> entityUsers = new HashSet<>();
+            Set<DotName> repos = new HashSet<>();
+            for (ClassInfo subclass : index.getIndex().getAllKnownImplementors(DotNames.PANACHE_REPOSITORY_BASE)) {
+                if (!subclass.name().equals(DotNames.PANACHE_REPOSITORY)) {
+                    repos.add(subclass.name());
+                }
+            }
+            for (ClassInfo implementor : index.getIndex().getAllKnownImplementors(DotNames.PANACHE_KOTLIN_REPOSITORY_BASE)) {
+                if (!implementor.name().equals(DotNames.PANACHE_KOTLIN_REPOSITORY)) {
+                    repos.add(implementor.name());
+                }
+            }
+            Set<DotName> entityReposUsers = new HashSet<>();
             for (DotName entity : entities) {
                 for (ClassInfo user : index.getIndex().getKnownUsers(entity)) {
-                    entityUsers.add(user.name());
+                    entityReposUsers.add(user.name());
+                }
+            }
+            for (DotName repo : repos) {
+                for (ClassInfo user : index.getIndex().getKnownUsers(repo)) {
+                    entityReposUsers.add(user.name());
                 }
             }
 
+            LOG.infof("entityReposUsers: %s\n\nentities: %s\n\nrepos: %s", entityReposUsers, entities, repos);
```

Comment:
I think we should change this method to `debugf` instead. Otherwise, avoid printing `entityReposUsers:` and write a more elegant text.